### DISCUSSION TITLE
Lolin S2 mini V1.0.0

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#11268] [ESP32]
Purchased a wemos S2 mini that supposedly had micropython installed as the default firmware. When I found it didn't I figured no problem I'll just load it. 

First problem no S2 V1.0.0 files @ https://github.com/wemos/micropython/releases. Found some likely contenders @ https://micropython.org/download/LOLIN_S2_MINI/  I wasn't sure what to make of [v1.19.1 (2022-06-18) .uf2] with its uf2 extension(seemed like a file for an rpi pico), but the [v1.18 (2022-01-17) .bin] looked plausible so downloaded & tried to flash.

Second problem. LOLIN_S2_MINI-20220117-v1.18.bin, A fatal error occurred: This chip is ESP32-S2 not ESP32. Wrong --chip argument? 

Can someone please point me to some upython firmware for S2 mini V1.0.0 that works?

**Comment by robert-hh** [comment]
Did you try the image from the official download site https://micropython.org/download/LOLIN_S2_MINI/ ?

**Reply by kjm1102** [reply]
Get the same error for all online files Rob. Fortunately when I stopped pressing the gpio0 button on power up & it identified on com16 instead of com17, there is micropython already on it so one less thing for me to worry about.

**Comment by scruss** [comment]
Lolin are usually pretty good at shipping a recent version of MicroPython. That's why I like buying from the official store.

Otherwise, I very recently had to flash an S2 mini with a new firmware. It doesn't use UF2 files, but bin files instead:

    esptool.py --chip esp32s2 --port /dev/ttyACM1 erase_flash
    esptool.py --chip esp32s2 --port /dev/ttyACM1 write_flash -z 0x1000 LOLIN_S2_MINI-20230413-unstable-v1.19.1-1015-g0acc73344.bin 

Substitute port and image name accordingly.
